app/assets/javascripts/editorToolbar.js

```diff
@@ -22,6 +20,22 @@ $(function() {
     $E[action](); // call the appropriate editor function
   });
 
+  // for save & recover buttons
+  $('.save-button').on('click', function(e) {
+    $E.setState(e.currentTarget.dataset.formId); // string that is: "main", "reply-123", "edit-123" etc.
+    $E.save($E);
```

Comment:
`$E.setState` in three new places: save buttons, recover buttons, clicking on textareas.